src.ts/providers/json-rpc-provider.ts

```diff
@@ -76,7 +72,15 @@ export class JsonRpcProvider extends Provider {
         const response = await fetchJson(this.connection, JSON.stringify(request));
         if (response.error) {
             if (typeof response.error.data === 'object') {
-                throw new TypedError(response.error.data.error_message, response.error.data.error_type);
+                if (typeof response.error.data.error_message === 'string' && typeof response.error.data.error_type === 'string') {
+                    // if error data has error_message and error_type properties, we consider that node returned an error in the old format
+                    throw new TypedError(response.error.data.error_message, response.error.data.error_type);
+                } else {
+                    // overwise we need to transform the new format into the old one for a transition period
+                    throw parseIntoOldTypedError(response.error.data);
```

Comment:
I think we actually want to have new errors but with messages masquerading for old errors. I'll do the change myself before merging.

Reply:
Sure, we can just add a new field to `TypedError` which will hold an error in a new format.